# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLImageElement.cpp
# Commit: 0a4882c4bafb
# Full Hash: 0a4882c4bafb451e05f5e8a4dcb671695e732ee9
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2022-08-27 17:34:11
# Description:
#   Bug 1787320 - Add the missing null check for mResponsiveSelector. r=emilio
#   
#   PictureSourceDimensionChanged may be called by an image element with a null
#   selector (because we call this function on all the image elements after
#   the source element).
# ==============================================================================

diff -r cb7738cd4618 -r 0a4882c4bafb dom/html/HTMLImageElement.cpp
--- a/dom/html/HTMLImageElement.cpp	Fri Aug 26 18:31:52 2022 +0000
+++ b/dom/html/HTMLImageElement.cpp	Fri Aug 26 18:45:12 2022 +0000
@@ -1028,7 +1028,7 @@
   // impact on the selection of <source> elements. In other words,
   // UpdateResponsiveSource doesn't change the source, so all we need to do is
   // just request restyle.
-  if (mResponsiveSelector->Content() == aSourceNode) {
+  if (mResponsiveSelector && mResponsiveSelector->Content() == aSourceNode) {
     InvalidateAttributeMapping();
   }
 }
